lighthouse-core/lib/i18n/i18n.js

```diff
@@ -174,7 +174,23 @@ const _icuMessageInstanceMap = new Map();
  */
 function _formatIcuMessage(locale, icuMessageId, icuMessage, values) {
   const localeMessages = LOCALES[locale];
-  const localeMessage = localeMessages[icuMessageId] && localeMessages[icuMessageId].message;
+  let localeMessage = localeMessages[icuMessageId] && localeMessages[icuMessageId].message;
+  // lets try to replace some things :o
+  const placeholders = localeMessages[icuMessageId] && localeMessages[icuMessageId].placeholders;
+  if (placeholders) {
+    console.log('placeholders!');
+    // do some regex
+    Object.entries(placeholders).forEach(entry => {
+      const key = entry[0];
+      const value = entry[1];
+      //use key and value here
+      let regexStr = `\$${key}\$`;
```

Comment:
I think we'll definitely want a `collect-strings` test that verifies that all the placeholders are in the string (would we also want the stricter check that there aren't any escaped values that don't have placeholders?)